### PR TITLE
ZCS-11673: Selected briefcase item get unselected when user set/unset Tag.

### PR DESCRIPTION
**Pull request by miteshsavani810**
- Remove selection of item, if user has uploaded new version of any selected item OR user has performed check-in/checkout operation on selected item.

### CODE REVIEW

File: WebRoot/js/zimbraMail/briefcase/controller/ZmBriefcaseController.js [hunk 1]
```diff
@@ -1416,7 +1416,26 @@ ZmBriefcaseController.prototype.handleModifyNotify =
...
-        view.deselectAll();
+        var removeSelection = false;
+        var selectedItems = view.getSelection();
+        var selecteItemsId = [];
+
+        for (var i = 0; i < selectedItems.length; i++) {
+            selecteItemsId.push(selectedItems[i].id);
+        }
+
+        for (var i = 0; i < modifies.doc.length; i++) {
```
Comment (k-kato) [inline WebRoot/js/zimbraMail/briefcase/controller/ZmBriefcaseController.js:1427]: Can we skip the `for` block if `selecteItemsId` is an empty array?
In addition, can we skip the process from line 1421 to 1438 if `selectedItems.length` is 0 ?

File: WebRoot/js/zimbraMail/briefcase/controller/ZmBriefcaseController.js [hunk 2]
```diff
@@ -1416,7 +1416,26 @@ ZmBriefcaseController.prototype.handleModifyNotify =
 function(modifies){
     var view = this._listView[this._currentViewId];
     if (view) {
-        view.deselectAll();
+        var removeSelection = false;
+        var selectedItems = view.getSelection();
+        var selecteItemsId = [];
+
+        for (var i = 0; i < selectedItems.length; i++) {
+            selecteItemsId.push(selectedItems[i].id);
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/briefcase/controller/ZmBriefcaseController.js:1424]: typo
```suggestion
            selectedItemsId.push(selectedItems[i].id);
```

File: WebRoot/js/zimbraMail/briefcase/controller/ZmBriefcaseController.js [hunk 3]
```diff
@@ -1416,7 +1416,26 @@ ZmBriefcaseController.prototype.handleModifyNotify =
...
+                removeSelection = true;
+                break;
+            }
+        }
+
+        // Unselect item when user upload item with new version of selected item or perform Checkout/Checkin on item.
+        if (removeSelection) {
+            view.deselectAll();
+        }
 	}
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/briefcase/controller/ZmBriefcaseController.js:1439]: indentation

File: WebRoot/js/zimbraMail/briefcase/controller/ZmBriefcaseController.js (hunk 1 above)
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/briefcase/controller/ZmBriefcaseController.js]: Fixed.

File: WebRoot/js/zimbraMail/briefcase/controller/ZmBriefcaseController.js (hunk 2 above)
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/briefcase/controller/ZmBriefcaseController.js]: Done.

File: WebRoot/js/zimbraMail/briefcase/controller/ZmBriefcaseController.js (hunk 3 above)
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/briefcase/controller/ZmBriefcaseController.js]: Fixed.

**Review by silentsakky — APPROVED** [review]

**Review by k-kato — APPROVED** [review]

**Pull request merged**